src/euphorie/client/browser/risk.py

```diff
@@ -530,8 +535,9 @@ def _getFile(self):
         if self.context.image_data is None:
             raise NotFound(self, self.fieldname, self.request)
 
-        if self.fieldname == "image_large":
-            image_data = self.get_or_create_image_scaled()
+        requested_scale = self.fieldname.partition("_")[-1] if self.fieldname else None
+        if requested_scale in self.allowed_sizes:
+            image_data = self.get_or_create_image_scaled(requested_scale)
```

Comment:
I do not think this is right because it has a race condition.
If you first request the "large" scale you will store it in `self.context.image_data_scaled`, then if you request the "preview" scale you will get the "large" one already stored in `self.context.image_data_scaled`.
Viceversa if you first request the "preview" and then the "large" you will get the "preview" one.
So I think we also need a different column to store a different preview.

With this in mind `image_data_scaled` seems a bad name...

Reply:
ok, then I will change it to just always use the "training" scale

Reply:
Your comment addresses exactly the problem that I wrote in my comment :)

Reply:
> ok, then I will change it to just always use the "training" scale
I would not hardcode the name.
I am thinking more to:
1. check if "requested_scale" if in the allowed sizes
2. check if we have a column where to store it
3. if the scale is allowed and we have a column return the column data or create the scale, otherwise return the unscaled image (or raise NotFound, depending on what you prefer)